### PR TITLE
Move Emoji to config.yaml, respect domain in config

### PR DESCRIPTION
**Pull request by aliasfalse**
Move emoji to config.yaml, makes it easier to start a self-hosted or adjust settings.

Respect the domain in config instead of hardcoding dashboard.rowboat.party

Previous defaults have been populated in config.example.yaml

### CODE REVIEW

File: config.example.yaml [hunk 1]
```diff
@@ -20,3 +20,34 @@ bot:
...
+  GREEN_TICK_EMOJI: 'green_tick:305231298799206401'
+  RED_TICK_EMOJI: 'red_tick:305231335512080385'
+  STATUS_ONLINE: ':status_online:305889169811439617',
+  STATUS_IDLE: ':status_away:305889079222992896',
+  STATUS_DND: ':status_dnd:305889053255925760',
+  STATUS_OFFLINE: ':status_offline:305889028996071425',
+  GAME_TYPE_STREAMING: ':status_streaming:305889126463569920',
+  
+rowboat_guild:
+  guild_id: 290923757399310337
```
Comment (nullpixel) [inline config.example.yaml]: fix casing here, should be `GUILD` or `GUILD_ID`

File: config.example.yaml [hunk 2]
```diff
@@ -20,3 +20,34 @@ bot:
...
+  role_id: 339256926921555968
+  
+SECRET_KEY: ''
+DSN: ''
+control_channels:
```
Comment (nullpixel) [inline config.example.yaml]: missing a space

File: config.example.yaml [hunk 3]
```diff
@@ -20,3 +20,34 @@ bot:
...
+  STATUS_IDLE: ':status_away:305889079222992896',
+  STATUS_DND: ':status_dnd:305889053255925760',
+  STATUS_OFFLINE: ':status_offline:305889028996071425',
+  GAME_TYPE_STREAMING: ':status_streaming:305889126463569920',
+  
+rowboat_guild:
+  guild_id: 290923757399310337
+  role_id: 339256926921555968
+  
+SECRET_KEY: ''
```
Comment (nullpixel) [inline config.example.yaml]: this should be in the `web:` section

File: config.example.yaml [hunk 4]
```diff
@@ -20,3 +20,34 @@ bot:
...
+
+web:
+   DOMAIN: 'https://dashboard.rowboat.party' # Domain for your instance. Leave out the trailing slash.
+   discord:
+     CLIENT_ID: '' # Your client ID from discordapp.com/developers
+     CLIENT_SECRET: '' # Your client secret from discordapp.com/developers
+     API_BASE_URL: 'https://discordapp.com/api'
+     TOKEN_URL: 'https://discordapp.com/api/oauth2/token'
+     AUTH_URL: 'https://discordapp.com/oauth2/authorize'
+     REDIRECT_URI: 'http://dahsboard.rowboat.party/auth/discord/callback' # Your apps callback URL for auth. This is yourdomain/auth/discord/callback
```
Comment (nullpixel) [inline config.example.yaml]: you spelt it wrong

File: rowboat/plugins/utilities.py [hunk 5]
```diff
@@ -24,6 +24,10 @@
 from rowboat.models.user import User, Infraction
 from rowboat.models.message import Message, Reminder
 from rowboat.util.images import get_dominant_colors_user, get_dominant_colors_guild
+from yaml import load
+
+with open('config.yaml', 'r') as f:
```
Comment (nullpixel) [inline rowboat/plugins/utilities.py]: you should not be opening the config repeatedly like this

File: rowboat/plugins/core.py [hunk 6]
```diff
@@ -29,17 +29,22 @@
 from rowboat.models.notification import Notification
 from rowboat.plugins.modlog import Actions
 
+from yaml import load
```
Comment (nullpixel) [inline rowboat/plugins/core.py]: why load this here when you have a `global_config` property in `CorePlugin` -- just move the emoji variables to a property

**Comment by aliasfalse** [comment]
There's actually a few additional parts... 
* Emoji config
* Dashboard link
* rmv-global-admin

**Comment by nullpixel** [comment]
I did actually have dashboard link, funnily enough you copied my exact typo.

**Comment by aliasfalse** [comment]
I was referring to the dashboard archive link.

**Comment by nullpixel** [comment]
so was I.

**Comment by aliasfalse** [comment]
honestly didn't read that part of yours. sorry.

**Comment by b1naryth1ef** [comment]
Closing this. I've extracted most of the constants that would need to be changed into a constants.py file. This can be edited for whatever local version you run. If you want to merge the domain stuff open a new PR with just that fix.

**Comment by aliasfalse** [comment]
ok

**Pull request closed without merging**